src/util/bip143.rs

```diff
@@ -168,15 +169,20 @@ impl<R: Deref<Target=Transaction>> SigHashCache<R> {
         })
     }
 
-    /// Compute the BIP143 sighash for any flag type. See SighashComponents::sighash_all simpler
-    /// API for the most common case
-    pub fn signature_hash(&mut self, input_index: usize, script_code: &Script, value: u64, sighash_type: SigHashType) -> SigHash {
-
+    /// Encode the BIP143 signing data for any flag type into a given object implementing a
+    /// std::io::Write trait.
+    pub fn encode_signing_data<Write: io::Write>(
```

Comment:
I'm not sure if `Write` should be first or last (which one is more idiomatic).

Also `enc` is not a best name`. Just `writer` would be better, IMO.

Maybe a `encode_signing_data_to` would be a better name. Maybe. Not sure.

Reply:
I agree on both of your suggestions regarding naming (`enc` -> `writer` and `encode_signing_data` -> `encode_signing_data_to`).

I'm not sure what is more idiomatic regarding the order of function parameters, but it makes more sense to me that `writer` becomes the first parameter if we rename the function. 

All in all, I'm fine with accepting all of your suggestions.
Thanks!

Reply:
Looks OK to me now. I'm more of a lurker around here, so you're going to need some more authoritative review.